dask/tests/test_utils.py

```diff
@@ -933,3 +940,56 @@ def test_get_meta_library_gpu():
     assert get_meta_library(cp.ndarray([])) == get_meta_library(
         da.from_array([]).to_backend("cupy")
     )
+
+
+@pytest.mark.parametrize(
+    "scheduler, expected_classes",
+    [
+        (None, ("SerializableLock", "SerializableLock", "AcquirerProxy")),
+        ("threads", ("SerializableLock", "SerializableLock", "SerializableLock")),
+        ("processes", ("AcquirerProxy", "AcquirerProxy", "AcquirerProxy")),
+    ],
+)
+def test_get_scheduler_lock(scheduler, expected_classes):
+    darr = da.ones((100,))
+    ddf = dd.from_dask_array(darr, columns=["x"])
+    dbag = db.from_sequence(np.ones((100,)), npartitions=2)
+
+    with mock.patch("distributed.Client") as client_class:
+
+        def no_client(*args, **kwargs):
+            raise ValueError("No client!")
+
+        client_class.current.side_effect = no_client
+
+        for collection, expected in zip((ddf, darr, dbag), expected_classes):
+            res = get_scheduler_lock(collection, scheduler=scheduler)
+            assert res.__class__.__name__ == expected
+
+
+@pytest.mark.parametrize(
+    "multiprocessing_method",
+    [
+        "spawn",
+        "fork",
+        "forkserver",
+    ],
+)
+def test_get_scheduler_lock_distributed(multiprocessing_method):
+    pytest.importorskip("distributed", reason="Requires distributed")
+
+    darr = da.ones((100,))
+    ddf = dd.from_dask_array(darr, columns=["x"])
+    dbag = db.from_sequence(np.ones((100,)), npartitions=2)
+
+    with dask.config.set(
+        {"distributed.worker.multiprocessing-method": multiprocessing_method}
```

Comment:
@graingert I split the tests for distributed into its own method. Independent of the `distributed.worker.multiprocessing-method`, I still get the same `AcquirerProxy` class though. I'm already returning `get_mp_context().Manager().Lock()`. Does this look right to you?